src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -85,9 +85,7 @@ private function configureRoutes(RoutingConfigurator $routes): void
             $routes->import($configDir.'/{routes}.php');
         }
 
-        if ($fileName = (new \ReflectionObject($this))->getFileName()) {
-            $routes->import($fileName, 'attribute');
-        }
+        $routes->import('routing.controllers');
```

Comment:
this makes any route defined on a class with the `routing.controller` tag win over routes defined in `routes.yaml`, while the existing solution respects the order of route definitions and so the current import might not be last. This could be a BC break if a project relies on such override.

And it also makes it impossible to override a route definition coming from a third-party bundle if that bundle defines a service tagged with `routing.controller`. This second issue could be solved by importing those *before* the `routes.yaml` (or `routes.php`) file though (but still after the `routes/*.yaml` files where recipes are importing third-party routes)

Reply:
I moved the import between `routes/*` and `routes.yaml`
Doesn't that solve everything?